frontend/tests/controllers/CertificatesTest.php

```diff
@@ -82,88 +78,34 @@ public function testConnectionRabbitMQ() {
+        );
+
+        // Create and add problems to the contest
         $problems = [];
         for ($i = 0; $i < 5; $i++) {
-            $problems[] = \OmegaUp\Test\Factories\Problem::createProblem();
-        }
+            $problems[$i] = \OmegaUp\Test\Factories\Problem::createProblem();
 
-        // Add problems to the contest
-        for ($i = 0; $i < 5; $i++) {
             \OmegaUp\Test\Factories\Contest::addProblemToContest(
                 $problems[$i],
                 $contestData
             );
         }
 
-        /// send submissions test1
-        for ($i = 0; $i < 5; $i++) {
-            $run = \OmegaUp\Test\Factories\Run::createRun(
-                $problems[$i],
-                $contestData,
-                $test1
-            );
-            \OmegaUp\Test\Factories\Run::gradeRun($run, 1.0, 'AC', 10);
-        }
-        /// send submissions test2
-        $run = \OmegaUp\Test\Factories\Run::createRun(
-            $problems[1],
-            $contestData,
-            $test2
-        );
-        \OmegaUp\Test\Factories\Run::gradeRun($run, 1.0, 'AC', 10);
-
-        /// send submissions test3
-        for ($i = 0; $i < 4; $i++) {
-            $run = \OmegaUp\Test\Factories\Run::createRun(
-                $problems[$i],
-                $contestData,
-                $test3
-            );
-            \OmegaUp\Test\Factories\Run::gradeRun($run, 1.0, 'AC', 10);
-        }
-
-        /// send submissions test4
-        $run = \OmegaUp\Test\Factories\Run::createRun(
-            $problems[2],
-            $contestData,
-            $test4
-        );
-        \OmegaUp\Test\Factories\Run::gradeRun($run, 1.0, 'AC', 10);
-
-        /// send submissions test5
-        $run = \OmegaUp\Test\Factories\Run::createRun(
-            $problems[3],
-            $contestData,
-            $test5
-        );
-        \OmegaUp\Test\Factories\Run::gradeRun($run, 1.0, 'AC', 10);
-
-        /// send submissions test6
-        $run = \OmegaUp\Test\Factories\Run::createRun(
-            $problems[0],
-            $contestData,
-            $test6
-        );
-        \OmegaUp\Test\Factories\Run::gradeRun($run, 1.0, 'AC', 10);
-
-        /// send submissions test7
-        for ($i = 0; $i < 2; $i++) {
-            $run = \OmegaUp\Test\Factories\Run::createRun(
-                $problems[$i],
-                $contestData,
-                $test7
-            );
-            \OmegaUp\Test\Factories\Run::gradeRun($run, 1.0, 'AC', 10);
-        }
-        /// send submissions test8
-        for ($i = 0; $i < 3; $i++) {
-            $run = \OmegaUp\Test\Factories\Run::createRun(
-                $problems[$i],
-                $contestData,
-                $test8
-            );
-            \OmegaUp\Test\Factories\Run::gradeRun($run, 1.0, 'AC', 10);
+        $submissions = [[0, 1, 2, 3, 4], [1], [0, 1, 2, 3], [2], [3], [0], [0, 1], [0, 1, 2]];
+
+        // Send submissions
+        for ($i = 0; $i < sizeof($submissions); $i++) {
```

Comment:
Acá podremos usar `foreach` en lugar de `for`
```suggestion
        foreach ($submissions as $index => $submission) {
```
?
Se ve mas entendible, además de que dejamos de utilizar la función para determinar el tamaño del arreglo.

Si te fijas, le puede agregar el indice, este te puede ser útil para saber a que usuario le vas a asignar cada envío